src/useFormo.ts

```diff
@@ -560,7 +560,7 @@ export function useFormo<
         subfield: subfieldName,
         errors: undefined,
       });
-      return success(values[name]);
+      return success((values as unknown as V)[name][index][subfieldName] as Values[K]);
```

Comment:
`as Values[K]` is needed because the function expects it to be that.

```ts
function validateSubfield<
    SK extends keyof SubFormValues<Values> & string,
    K extends SubFormKeys<Values>,
    V extends ArrayRecord<Values, SK>
  >(
    name: K,
    index: number,
    subfieldName: SK,
    values: Values
  ): Promise<Result<NonEmptyArray<FieldError>, Values[K]>>
```
The return type should be something like "`Promise<Result<NonEmptyArray<FieldError>, ArrayItem<Values[K]>[SK]>>`" as that's what we're actually returning.

Reply:
ah uhm, I see what you mean, `Values[K]` does not seem right (still it is being used and casted to in a lot of other places)